src/main/java/com/google/firebase/database/tubesock/ThreadConfig.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.database.tubesock;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.firebase.database.core.ThreadInitializer;
+
+import com.google.firebase.internal.NonNull;
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * Configuration that governs how the websocket connections spawn and initialize threads
+ * for reading and writing data.
+ */
+public final class ThreadConfig {
+
+  private final ThreadInitializer threadInitializer;
+  private final ThreadFactory threadFactory;
+
+  public ThreadConfig(ThreadInitializer threadInitializer, ThreadFactory threadFactory) {
```

Comment:
Nit: Swap the order here. First thread get 'factoried' and then initialized.

Reply:
Done